### PR TITLE
Add country code to the operator number

### PR DESCRIPTION
**Pull request by ostap0207**
With adding EU numbers we have to add country code to the numbers. This affects out public api.

LUF-71

### CODE REVIEW

**Comment by deiwin** [comment]
I'm not sure we can do this. This is a backwards incompatible change.

**Comment by ostap0207** [comment]
@indrekj @urmastalimaa what do you think about this change. It might break the API consumers code though. Other solution could be to store the `phone code` separately, which is harder to support and also might break clients' code.

**Comment by ostap0207** [comment]
@deiwin I am aware, just wanted to start discussion.

**Comment by indrekj** [comment]
@ostap0207 can you check from logasm how and who is using these endpoints? Maybe we can change it.

**Comment by justindipietro** [comment]
If it's being used it's currently only the US - perhaps that helps in
adding a 1 if it's not there / given

I know creating operators is being used by wfg

On Tuesday, April 5, 2016, Indrek Juhkam <notifications@github.com> wrote:

> @ostap0207 <https://github.com/ostap0207> can you check from logasm how
> and who is using these endpoints? Maybe we can change it.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/salemove/apiary_docs/pull/27#issuecomment-205733894>
>


-- 

*Justin DiPietro*
Co-founder & COO | SaleMove
19 W 21st St. Suite 202, New York, NY, 10010
+1.732.266.3854 | SaleMove | www.salemove.com
<http://salemove.com/#sm_show_notification?message=justin+email+signature+link&note=clicked+on+justin+email+signature>
http://linkedin.com/in/justindipietro

___________________________________________________________________

This email and any files transmitted with it are confidential and
intended solely for the use of the individual or entity to whom they
are addressed. If you have received this email in error, please notify
the system manager. This message contains confidential information and
is intended only for the individual named. If you are not the named
addressee, you should not disseminate, distribute or copy this email.
Please notify the sender immediately by email if you have received
this email by mistake and delete this email from your system. If you
are not the intended recipient, you are notified that disclosing,
copying, distributing or taking any action in reliance on the contents
of this information is strictly prohibited.

**Comment by deiwin** [comment]
The issue with just prepending the `+1` to the current output is that the clients might also be doing the same, in which case they're numbers would now be in the form of `+1+1987654321`.

**Comment by ostap0207** [comment]
Yeah, can see from the logs the it's being used.

**Comment by ostap0207** [comment]
as @justindipietro said in the US they are probably not adding +1. The issue can be with formatting if they have custom UI.

**Comment by justindipietro** [comment]
Can we make a check for country code? If no country code is there we assume
it's from the US and add a +1?

On Tuesday, April 5, 2016, Ostap <notifications@github.com> wrote:

> as @justindipietro <https://github.com/justindipietro> said in the US
> they are probably not adding +1. The issue can be with formatting if they
> have custom UI.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/salemove/apiary_docs/pull/27#issuecomment-205738674>
>


-- 

*Justin DiPietro*
Co-founder & COO | SaleMove
19 W 21st St. Suite 202, New York, NY, 10010
+1.732.266.3854 | SaleMove | www.salemove.com
<http://salemove.com/#sm_show_notification?message=justin+email+signature+link&note=clicked+on+justin+email+signature>
http://linkedin.com/in/justindipietro

___________________________________________________________________

This email and any files transmitted with it are confidential and
intended solely for the use of the individual or entity to whom they
are addressed. If you have received this email in error, please notify
the system manager. This message contains confidential information and
is intended only for the individual named. If you are not the named
addressee, you should not disseminate, distribute or copy this email.
Please notify the sender immediately by email if you have received
this email by mistake and delete this email from your system. If you
are not the intended recipient, you are notified that disclosing,
copying, distributing or taking any action in reliance on the contents
of this information is strictly prohibited.

**Comment by deiwin** [comment]
Yeah, more generally, the implicit (I think, or is it explicit?) specification that all numbers are US ones without the country code will be broken. Anything that depends on that will break.

**Comment by ostap0207** [comment]
@justindipietro That can be done. The same has to be done than when someone fetches the operator info. If the number had +1 remove it. Sounds hacky though. Special status for the US :)

**Comment by ostap0207** [comment]
Twilio right now has something like this. If number doesn't have the, they assume it's from the US.

**Comment by justindipietro** [comment]
Perfect- but ya the US gets special treatment :) most of our clients are US
so it makes sense

On Tuesday, April 5, 2016, Ostap <notifications@github.com> wrote:

> Twilio right now has something like this. If number doesn't have the, they
> assume it's from the US.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/salemove/apiary_docs/pull/27#issuecomment-205743800>
>


-- 

*Justin DiPietro*
Co-founder & COO | SaleMove
19 W 21st St. Suite 202, New York, NY, 10010
+1.732.266.3854 | SaleMove | www.salemove.com
<http://salemove.com/#sm_show_notification?message=justin+email+signature+link&note=clicked+on+justin+email+signature>
http://linkedin.com/in/justindipietro

___________________________________________________________________

This email and any files transmitted with it are confidential and
intended solely for the use of the individual or entity to whom they
are addressed. If you have received this email in error, please notify
the system manager. This message contains confidential information and
is intended only for the individual named. If you are not the named
addressee, you should not disseminate, distribute or copy this email.
Please notify the sender immediately by email if you have received
this email by mistake and delete this email from your system. If you
are not the intended recipient, you are notified that disclosing,
copying, distributing or taking any action in reliance on the contents
of this information is strictly prohibited.

**Comment by deiwin** [comment]
@ostap0207, can you please describe more exactly what you're planning to do. Or what it is that `[..] can be done`.

**Comment by ostap0207** [comment]
`That can be done` is a reply to Justin's suggestion https://github.com/salemove/apiary_docs/pull/27#issuecomment-205740080. It means 
1) we still accept numbers without the code. In this case we add `+1` internally.
2) when someone fetches the operator we delete `+1` form the number if it exists.
This would allow to keep the backward compatibility and add new number codes for other part of the world.

**Comment by ostap0207** [comment]
The same probably has to be done for visitor

**Comment by deiwin** [comment]
> This would allow to keep the backward compatibility [..]

This is only true if we can guarantee no one is currently prepending `+1` to the numbers.

**Comment by justindipietro** [comment]
@deiwin I think its a safe assumption that no one is currently prepending +1 to the numbers at this point. Most people in the US only consider the US :)

**Comment by deiwin** [comment]
Fair enough. Under this assumption, I think the proposed approach is good enough. @ostap0207, the commit has to be updated though, to remove the `+1` from the responses and to describe the country code omission for US.

**Comment by ostap0207** [comment]
Updated

File: apiary.apib [hunk 1]
```diff
@@ -1362,12 +1362,15 @@ To create a new operator, you must provide a unique email address.
             Authorization: Token 2pFuPlTYkH7T4MylrwxApA
             Accept: application/vnd.salemove.v1+json
 
+    + Attributes
+        + phone (string) - phone number formatted in E.164. US phone numbers are accepted without the calling code '+1'.
```
Comment (deiwin) [inline apiary.apib]: I think it should be either "country calling code" or "country code", but not just "calling code".

Comment (deiwin) [inline apiary.apib]: Elsewhere as well

File: apiary.apib [hunk 2]
```diff
@@ -1380,11 +1383,16 @@ To create a new operator, you must provide a unique email address.
 
 + Response 200 (application/json)
 
+    + Attributes
+        + phone (string) - phone number formatted in E.164. In the case of US number, calling code '+1' is removed from the number.
```
Comment (deiwin) [inline apiary.apib]: > In the case of US number, calling code '+1' is removed from the number.

Perhaps `For US numbers, the country calling code '+1' is omitted.` reads better.

**Comment by ostap0207** [comment]
Updated

**Comment by urmastalimaa** [comment]
+1

**Comment by deiwin** [comment]
Is the restriction of the length of 10 for US numbers so obvious that it shouldn't be specified here?

**Comment by ostap0207** [comment]
Let's ask @justindipietro about that.

**Comment by ostap0207** [comment]
Those endpoints are used now with correct 10 digit numbers though.

**Comment by ostap0207** [comment]
@justindipietro could you please respond to Deiwin's comment https://github.com/salemove/apiary_docs/pull/27#issuecomment-206752505.

**Comment by justindipietro** [comment]
In the US it is pretty obvious that we have 10 digest numbers. Though
sometimes you will see an 11 digest if the person adds a 1 for country code

90% of the time- 732-266-3854
10% of the time- 1-732-266-3854

On Tuesday, April 12, 2016, Ostap <notifications@github.com> wrote:

> @justindipietro <https://github.com/justindipietro> could you please
> respond to Deiwin's comment #27 (comment)
> <https://github.com/salemove/apiary_docs/pull/27#issuecomment-206752505>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/salemove/apiary_docs/pull/27#issuecomment-208828888>
>


-- 
Sent from my iPhone

**Comment by ostap0207** [comment]
So I think having specified that a US number should be without `+1` is enough here.

**Comment by deiwin** [comment]
I mean, it can be with +1 as well, because it'll be in the E.164 format then, no?

**Comment by ostap0207** [comment]
Yeah, it still can be with `+1`. We could say `US phone numbers are *also* accepted without the country calling code '+1'` or `US phone numbers are  accepted both with or without the country calling code '+1'`

**Comment by deiwin** [comment]
I think the current description is fine. If you want, you can add the `also`, but either way is fine for me.

**Pull request merged**